src/linter.js

```diff
@@ -629,4 +634,25 @@ export default class Linter {
       });
   }
 
+  _markBadwordUsage(filename, fileData) {
+    if (fileData) {
+      const badwordsRe = new RegExp(badwords.en.join('|'), 'gi');
+
+      const words = fileData.split(' ');
+
+      for (let word of words) {
+        // Filter out none a-Z characters to simplify matching with our
+        // badword list.
+        word = word.toLowerCase().replace(/[^a-zA-Z]/, '');
+
+        if (word.match(badwordsRe)) {
```

Comment:
I haven't, let me do that and get back to you on that. I'm not too worried though tbh, we are iterating through the AST quite a few times during the linting process which imho should be slower even than regexes.

Especially line 639 shouldn't be too slow imho. I'll bechmark that though with a few bigger add-ons and see if it has performance implications.